Add the translator nudge for non-Mag-16 locales

d-alleyne opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to Automattic/i18n-issues#850

## Proposed Changes

Adds a translation nudge on the Calypso home screen, and to the language switcher (in next commit)

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

We removed the nudge on https://github.com/Automattic/wp-calypso/pull/91826, but this re-adds it so that it is more targeted 

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Login into Calypso, and change the language to one whose translation percentage is less than 85%.
- Verify that the nudge is shown to the users

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

d-alleyne committed:
Add Material Emoji Lanaguage icon

d-alleyne committed:
Add minor text

d-alleyne committed:
Build home widget to display the new notice

d-alleyne committed:
Add translation nudge to the home page

d-alleyne committed:
Remove unused methods

yuliyan commented on client/components/translator-invite/index.jsx:
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
+			location: this.props.path,
 		} );
 
+	renderIcon() {
```
This method is currently unused.

d-alleyne replied:
Thanks for taking the time to review. 
A local commit had removed the unused methods. Implementing the Language switcher version is in progress.
> There are discrepancies with https://github.com/Automattic/i18n-issues/issues/850#issuecomment-2231355732 with regards to the margins, font sizes, and colors.

Good eye. I was mainly verifying that the layout and relative sizes were correct.

> I don't really see a major benefit of using a single component, so I think using two separate components would allow to drop some of the logic and might make the implementation a little easier to follow.

Will do.

yuliyan commented on client/components/translator-invite/index.jsx:
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
+		return <MaterialIcon icon="emoji_language" />;
+	}
+
+	renderHeader() {
```
This method is currently unused.

yuliyan commented on client/components/translator-invite/index.jsx:
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
+		const { locale, translate } = this.props;
+		const link = (
+			<a
+				className="language-picker__modal-incomplete-locale-notice-learn-more"
```
According to the [BEM naming convention](https://github.com/Automattic/wp-calypso/blob/ce1a376af4bcc8987855ced4c961efacda6e1d32/packages/eslint-plugin-wpcalypso/docs/rules/jsx-classname-namespace.md?plain=1#L3), the class name here suggest that the link is an element of `language-picker`, whereas it's actually an element of `translator-invite`.

yuliyan commented on client/components/translator-invite/index.jsx:
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
 
-		return null;
+		return (
+			<div className="language-picker__modal-incomplete-locale-notice-info">{ noticeText }</div>
```
According to the [BEM naming convention](https://github.com/Automattic/wp-calypso/blob/ce1a376af4bcc8987855ced4c961efacda6e1d32/packages/eslint-plugin-wpcalypso/docs/rules/jsx-classname-namespace.md?plain=1#L3), the class name here suggest that the notice text is an element of `language-picker`, whereas it's actually an element of `translator-invite`.

d-alleyne replied:
Aah, I was unaware that we were following a specific convention. I was reusing [the classes](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/language-picker/modal.tsx#L80) found in the Language Switcher when developing this component.

yuliyan commented on client/my-sites/customer-home/cards/features/app-promo/index.tsx:
```diff
@@ -1,15 +1,21 @@
+				) }
+				campaign="calypso-customer-home"
+			/>
+			<TranslatorInvite path="/home" />
```
Is there a reason to render the `TranslatorInvite` as part of the `AppPromoHome` card instead of rendering it as its own card?

d-alleyne replied:
Initially, that was the plan. However, making a new card [appear on the Calypso Home page](https://github.com/Automattic/wp-calypso/blob/1384afc6779ff93ac95d9ebc443d93a9b6579b6d/client/my-sites/customer-home/locations/tertiary/manage-site/index.jsx#L4) was more complex than I expected. It was simpler to add the component to the AppPromoHome.

yuliyan reviewed (commented):
The translator nudge messages intended for the site home screen and the language picker, seem to have more differences than similarities. Using a single component to fit both use cases seems to have resulted in adding unnecessary complexity to a component that's supposed to simply render an icon with some text and a link.

I don't really see a major benefit of using a single component, so I think using two separate components would allow to drop some of the logic and might make the implementation a little easier to follow.

----

Apart from that, I've noticed some other problems:

* Some unit tests are failing.
* There are discrepancies with [the proposed the design](https://github.com/Automattic/i18n-issues/issues/850#issuecomment-2231355732) with regards to the margins, font sizes, and colors.
* The `TranslatorInvite` is currently not being used on the language picker, despite having the implementation for it.

yuliyan commented on client/components/translator-invite/index.jsx:
```diff
@@ -28,50 +28,110 @@ export class TranslatorInvite extends Component {
+				return (
+					<>
+						<MaterialIcon icon="emoji_language" />
+						<p className="card-heading card-heading-20">
```
Using the class names from [the CardHeading component](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/card-heading/index.jsx#L12) will only provide the expected styling if the `CardHeading` is used elsewhere in the current client session so that it's stylesheet gets loaded, which may not always be the case.

d-alleyne committed:
Implement code review suggestions

- Follow BEM guidelines
- Remove the component out of the AppPromo component

d-alleyne committed:
Add the translator invite on the Language Picker

yuliyan commented on client/components/language-picker/modal.scss:
```diff
@@ -67,6 +67,19 @@
+
+	.material-icon {
+		margin: 0 2px -4px 0;
+		transform: scale(0.8);
```
The [<MaterialIcon /> component accepts `size` prop](https://github.com/Automattic/wp-calypso/blob/d2f640d00a8f4b4782dca1b9153454ed3fc30c43/packages/components/src/material-icon/index.tsx#L13). It's preferred to use that instead of scaling down the icon via `transform`.

d-alleyne replied:
The `size` prop depends on the `.svg` files of that [exact name](https://github.com/Automattic/wp-calypso/blob/e5b49b40406c4a8adc844e3fbab8c57d13e9e462/packages/components/src/material-icon/index.tsx#L20) existing [material-design-icons](https://github.com/Automattic/wp-calypso/blob/trunk/packages/material-design-icons/README.md) directory.

yuliyan replied:
Ah, I didn't see that. In this case, the `width` and `height` could be set in the CSS.

yuliyan commented on client/components/language-picker/modal.scss:
```diff
@@ -67,6 +67,19 @@
+}
+
+.language-picker__modal-incomplete-locale-nudge-link {
+	font-weight: normal;
```
Not: The `normal` seems to be inherited from the default styles, so it seems redundant to set it explicitly here.

yuliyan commented on client/components/translator-invite/style.scss:
```diff
@@ -1,20 +1,26 @@
-.is-section-signup .translator-invite__content a {
-	color: var(--color-text-inverted);
-}
+	&.notice-header {
```
Since we've [previously mentioned](https://github.com/Automattic/wp-calypso/pull/92772#discussion_r1691070010) the [BEM naming convention](https://en.bem.info/methodology/naming-convention/), here's an example of the class names that I would go with to be compliant with it:

```css
.translator-invite__content { ... }
.translator-invite__heading { ... }
.translator-invite__subheading { ... }
.translator-invite__link { ... }
.translator-invite__icon { ... }
```

yuliyan commented on client/my-sites/customer-home/cards/features/app-promo/index.tsx:
```diff
@@ -4,12 +4,16 @@ import AppPromo from 'calypso/blocks/app-promo';
-			subheader={ translate( 'Everything you need to build and grow your site from any device.' ) }
-			campaign="calypso-customer-home"
-		/>
+		<>
```
I believe the fragment here is not required. The formatting changes also seem unrelated to this PR.

yuliyan commented on client/my-sites/customer-home/locations/tertiary/manage-site/index.jsx:
```diff
@@ -63,6 +64,7 @@ const ManageSite = () => {
 						key: index,
 					} )
 			) }
+			<TranslatorInvite path="/home" />
```
Would it make sense to move one level up, directly inside [the Home `<Tertiary />` location](https://github.com/Automattic/wp-calypso/blob/a56204b117473f4379ac0e8039adc2aea95e7406/client/my-sites/customer-home/locations/tertiary/index.jsx#L23)?

yuliyan reviewed (changes requested):
### Visual remarks

#### Home screen

![vxEFZmY0n331EBRA](https://github.com/user-attachments/assets/ef4553f7-f966-4245-b978-27e97126dd28)

1. The icon color is different from the design.
2. The margin between the two cards doesn't match.
3. Font-weight is different between the two cards. The link color is different from the design, but matches other links on the screen so I'm not sure which one should be used. The link underline has more offset in the design and the card above.
4. Missing padding on mobile viewport:
![OPnBNkPD6lnRK7aV](https://github.com/user-attachments/assets/466245c9-60ae-4485-a887-ae4c2e85f45f)

#### Language picker

![sidDlNp4RCaUPmrV](https://github.com/user-attachments/assets/01b25365-8f68-4f99-95a9-402e104389bd)

1. The icon is not aligned with the checkbox above it.
2. The color of the icon doesn't match the design.
3. There's less space between the percentage notice and the translator nudge.
4. I don't think it's ok to have the notice inside a label element: 
![QlVFlS7hkbO2Yagy](https://github.com/user-attachments/assets/36bb27c0-70e2-4ccd-992f-d105255e1a81)



### Other

* Unit tests are failing. Could you fix them?
* Since the components visibility is based on the translated percentage, it might be worth to follow up updating the languages data, as it seems like the [languages packages hasn't been updated in a while](https://github.com/Automattic/wp-calypso/blob/trunk/packages/languages/src/languages-meta.json).

d-alleyne committed:
Fix issues with the TranslatorInvite component

- Icon color changed to match design
- Margin between description and link updated.
- Font weight, and link color, and link color underline offset match
- Added padding to mobile view

d-alleyne committed:
Implement code review suggestions

d-alleyne committed:
Update the languages package

d-alleyne committed:
Use BEM class names

d-alleyne committed:
Minor CSS changes

d-alleyne committed:
Fix tests, and code cleanup

d-alleyne committed:
Use isTranslatedCompletely rather than translation percentages

yuliyan commented on client/components/language-picker/modal.scss:
```diff
@@ -67,6 +67,17 @@
+.language-picker__modal-incomplete-locale-nudge-text {
+	font-weight: normal;
+	/* stylelint-disable-next-line scales/font-sizes */
+	font-size: 0.9375rem; //typography-exception
```
The `language-picker__modal-incomplete-locale-text` element above has a smaller font size of `0.875rem`. Let's use the same one here for consistency. It would also make the disabling the styleling rule obsolete.

yuliyan commented on client/components/language-picker/modal.tsx:
```diff
@@ -144,6 +135,21 @@ const LanguagePickerModal: React.FC< Props > = ( {
 							</span>
 						</FormLabel>
 					) }
+					{ selectedLanguage.isTranslatedIncompletely && (
```
The `selectedLanguage.isTranslatedIncompletely` seems redundant as it's already [one of the condition to render the parent element](https://github.com/Automattic/wp-calypso/pull/92772/files#diff-e2c927e8e979f0f2d70ee47e9b25071d8e3232a3c9a1842993925b8dfb5c8c11L125).

d-alleyne replied:
Good catch!

yuliyan commented on client/components/translator-invite/index.jsx:
```diff
@@ -28,50 +29,63 @@ export class TranslatorInvite extends Component {
+						{ this.renderNoticeLabelText() }
+
+						<a
+							className="translator-invite__link is-link"
```
The `is-link` class name doesn't seem to provide any styling, so I think it's not needed.

yuliyan commented on client/components/translator-invite/index.jsx:
```diff
@@ -28,50 +29,63 @@ export class TranslatorInvite extends Component {
 		return (
 			<div className="translator-invite">
-				{ this.renderNoticeLabelText() }
+				{ isTranslatedIncompletely && (
```
The component will render the empty `translator-invite` div and the `<QueryLanguageNames />` for languages with complete languages, which I suppose is not intended.

d-alleyne replied:
Actually, this was done to preserve existing behaviour, and fix [this test](https://github.com/Automattic/wp-calypso/blob/8464223a66e00048511ce73b27f39e02093a1742/client/components/translator-invite/test/index.js#L37).

yuliyan replied:
I don't really see the point to render an empty DOM element in this case. Also, requesting the language names data that is not going to be used because the rendering condition is not fulfilled seems pointless too.

yuliyan commented on client/components/translator-invite/style.scss:
```diff
@@ -1,20 +1,32 @@
-	margin: 15px auto;
-	font-size: $font-body-extra-small;
-	font-style: italic;
 	max-width: 75%;
```
Is the max width property left out intentionally?

yuliyan commented on client/components/translator-invite/style.scss:
```diff
@@ -1,20 +1,32 @@
 }
 
-.is-section-signup .translator-invite__content a {
+.translator-invite__link {
```
The font size should be `0.9375rem` and font weight `600` in order to match the styling in the app promo card above.

yuliyan commented on packages/material-design-icons/src/communication/outline-emoji_language-16px.svg:
Is there any other difference than the size? If not, then the original icon can be resized with CSS where needed instead of introducing a separate SVG.

d-alleyne committed:
Implement code suggestions

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16458391

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @d-alleyne, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `%(languageName)s is only %(percentTranslated)d%% translated. Help translate WordPress into your language.`
- `Translate WordPress.com`

Thank you in advance!

d-alleyne committed:
Resolve merge conflicts

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16458391).
